### PR TITLE
sched/signal: move signal structures pool to bss

### PR DESCRIPTION
**Pull request by anchao**
## Summary

sched/signal: move signal structures pool to bss
sched/mqueue: move message queue buffer to bss

Decoupling with memory allocator

## Impact

N/A

## Testing

ostest

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: sched/mqueue/mq_initialize.c [hunk 1]
```diff
@@ -52,6 +52,22 @@ struct list_node g_msgfreeirq;
 
 #endif
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* This is a pool of pre-allocated message queue buffers */
+
+static uint8_t g_msgpool[
```
Comment (xiaoxiang781216) [inline sched/mqueue/mq_initialize.c:61]: the alignment mayn't match the requirementt of mqueue_msg_s/msgbuf_s

File: sched/signal/sig_initialize.c [hunk 2]
```diff
@@ -68,6 +68,17 @@ sq_queue_t  g_sigpendingsignal;
 
 sq_queue_t  g_sigpendingirqsignal;
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* This is a pool of pre-allocated signal structures buffers */
+
+static uint8_t g_sigpool[
```
Comment (xiaoxiang781216) [inline sched/signal/sig_initialize.c:77]: dittto

File: boards/arm/lpc17xx_40xx/lpc4088-devkit/scripts/memory.ld [hunk 3]
```diff
@@ -73,9 +73,9 @@ MEMORY
 
   /* 64Kb of SRAM in the CPU block */
 
-  ksram (rwx)      : ORIGIN = 0x10000000, LENGTH = 4K   /* May include waste */
-  usram (rwx)      : ORIGIN = 0x10001000, LENGTH = 4K
-  xsram (rwx)      : ORIGIN = 0x10002000, LENGTH = 24K  /* All used as heap */
+  ksram (rwx)      : ORIGIN = 0x10000000, LENGTH = 6K   /* May include waste */
```
Comment (xiaoxiang781216) [inline boards/arm/lpc17xx_40xx/lpc4088-devkit/scripts/memory.ld:76]: why need

File: sched/mqueue/mq_initialize.c (hunk 1 above)
Comment (anchao) [reply sched/mqueue/mq_initialize.c]: Done

File: sched/signal/sig_initialize.c (hunk 2 above)
Comment (anchao) [reply sched/signal/sig_initialize.c]: Done

File: boards/arm/lpc17xx_40xx/lpc4088-devkit/scripts/memory.ld (hunk 3 above)
Comment (anchao) [reply boards/arm/lpc17xx_40xx/lpc4088-devkit/scripts/memory.ld:76]: bss section out of range

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**